agents/agents-features/agents-features-acp/src/jvmMain/kotlin/ai/koog/agents/features/acp/MessageConverters.kt

```diff
@@ -0,0 +1,205 @@
+package ai.koog.agents.features.acp
+
+import ai.koog.prompt.message.AttachmentContent
+import ai.koog.prompt.message.ContentPart
+import ai.koog.prompt.message.Message
+import ai.koog.prompt.message.RequestMetaInfo
+import com.agentclientprotocol.common.Event.SessionUpdateEvent
+import com.agentclientprotocol.model.ContentBlock
+import com.agentclientprotocol.model.EmbeddedResourceResource
+import com.agentclientprotocol.model.SessionUpdate
+import com.agentclientprotocol.model.SessionUpdate.AgentMessageChunk
+import com.agentclientprotocol.model.ToolCallId
+import com.agentclientprotocol.model.ToolCallStatus
+import kotlinx.datetime.Clock
+
+public const val UNKNOWN_FORMAT: String = "unknown"
+public const val UNKNOWN_MIME_TYPE: String = "unknown/unknown"
+public const val UNKNOWN_URI: String = "unknown"
+public const val UNKNOWN_FILE_NAME: String = "unknown"
+public const val UNKNOWN_TOOL_CALL_ID: String = "unknown"
+public const val UNKNOWN_TOOL_DESCRIPTION: String = "unknown"
+
+private fun parseFormat(mimeType: String?): String {
+    return mimeType?.split("/")?.lastOrNull() ?: UNKNOWN_FORMAT
+}
+
+/**
+ * Converts a list of [ContentBlock] of ACP prompt to a Koog [Message.User].
+ */
+public fun List<ContentBlock>.toKoogMessage(clock: Clock): Message {
+    return Message.User(
+        parts = this.map { it.toKoogContentPart() },
+        metaInfo = RequestMetaInfo(clock.now())
+    )
+}
+
+/**
+ * Converts a single [ContentBlock] of ACP prompt to a Koog [ContentPart].
+ */
+public fun ContentBlock.toKoogContentPart(): ContentPart {
+    return when (this) {
+        // https://agentclientprotocol.com/protocol/content#audio-content
+        is ContentBlock.Audio -> {
```

Comment:
Here and for `Image` you can put `uri` from ACP's entities into our `fileName` field. Technically it's not really correct, but at least we will preserve this information somehow, rather than just leaving it as an unnamed file